nvidia_gpu_stats/src/gpu_nvidia.rs

```diff
@@ -0,0 +1,252 @@
+/// gpu.{i}.fanSpeed: The current fan speed of the GPU at index i (in percentage).
+/// gpu.{i}.encoderUtilization: The utilization of the GPU's encoder at index i (in percentage).
+/// gpu.{i}.gpu: The overall GPU utilization at index i (in percentage).
+/// gpu.{i}.memory: The GPU memory utilization at index i (in percentage).
+/// gpu.{i}.memoryTotal: The total memory of the GPU at index i (in bytes).
+/// gpu.{i}.memoryAllocated: The percentage of GPU memory allocated at index i.
+/// gpu.{i}.memoryAllocatedBytes: The amount of GPU memory allocated at index i (in bytes).
+/// gpu.{i}.temp: The temperature of the GPU at index i (in Celsius).
+/// gpu.{i}.powerWatts: The power consumption of the GPU at index i (in Watts).
+/// gpu.{i}.enforcedPowerLimitWatts: The enforced power limit of the GPU at index i (in Watts).
+/// gpu.{i}.powerPercent: The percentage of power limit being used by the GPU at index i.
+/// gpu.{i}.graphicsClock: The current graphics clock speed of the GPU at index i (in MHz).
+/// gpu.{i}.memoryClock: The current memory clock speed of the GPU at index i (in MHz).
+/// gpu.{i}.pcieLinkGen: The current PCIe link generation of the GPU at index i.
+/// gpu.{i}.pcieLinkSpeed: The current PCIe link speed of the GPU at index i (in bits per second).
+/// gpu.{i}.pcieLinkWidth: The current PCIe link width of the GPU at index i.
+/// gpu.{i}.maxPcieLinkGen: The maximum PCIe link generation supported by the GPU at index i.
+/// gpu.{i}.maxPcieLinkWidth: The maximum PCIe link width supported by the GPU at index i.
+/// gpu.{i}.cudaCores: The number of CUDA cores in the GPU at index i.
+/// gpu.{i}.architecture: The architecture of the GPU at index i (e.g., Ampere, Turing).
+/// gpu.process.{i}.*: Various metrics specific to the monitored process
+///    (if the GPU is in use by the process). These include GPU utilization, memory utilization,
+///     temperature, and power consumption.
+/// _timestamp: The Unix timestamp when the metrics were collected.
+///
+/// Note that {i} represents the index of each GPU in the system, starting from 0.
+pub struct NvidiaGpu {
+    nvml: Nvml,
+    cuda_version: String,
+    device_count: u32,
+}
+
+impl NvidiaGpu {
+    pub fn new() -> Result<Self, NvmlError> {
+        let nvml = Nvml::init()?;
+        let cuda_version = nvml.sys_cuda_driver_version()?;
+        format!(
+            "{}.{}",
+            nvml_wrapper::cuda_driver_version_major(cuda_version),
+            nvml_wrapper::cuda_driver_version_minor(cuda_version)
+        );
+        let device_count = nvml.device_count()?;
+
+        Ok(NvidiaGpu {
+            nvml,
+            cuda_version: format!(
+                "{}.{}",
+                nvml_wrapper::cuda_driver_version_major(cuda_version),
+                nvml_wrapper::cuda_driver_version_minor(cuda_version)
+            ),
+            device_count,
+        })
+    }
+
+    /// Function to check if a GPU is being used by a specific process or its children
```

Comment:
The doc doesn't need to mention that this is a function. There's a short story that's relevant here: http://www.emotionalintelligenceatwork.com/resources/fresh-fish-sold-daily/

also please add a period at the end, otherwise it looks like it might be an unfinished sentence (which we have merged before)

Reply:
indeed, this function works even on Sabbath.